cartographer/mapping/global_trajectory_builder_interface.h

```diff
@@ -58,6 +58,15 @@ class GlobalTrajectoryBuilderInterface {
   virtual void AddSensorData(const sensor::OdometryData& odometry_data) = 0;
   virtual void AddSensorData(
       const sensor::FixedFramePoseData& fixed_frame_pose) = 0;
+
+  // A callback which is called after local SLAM processes an accumulated
+  // 'sensor::RangeData'. If the data was inserted into a submap, reports the
+  // assigned 'NodeId', otherwise 'nullptr' if the data was filtered out.
+  using AccumulatedRangeDataCallback =
```

Comment:
The range data is only part of what this callback gets as information. `LocalSlamResultCallback` or `LocalSlamCallback` maybe? This would also better match your comment. @cschuet Opinion?

Reply:
I would prefer `LocalSlamResultCallback` but both are fine for me.

Reply:
The "range data" in the callback name refers not only to the contained range data that is passed to the callback, but to the fact that the callback has been called right after this data has been processed (the callback is called from `AddRangefinderData()` after all). The callback is not called after e.g. adding IMU or odometry data, so I thought this name would emphasize this.

But the question is - is that important? The user probably just cares about the pose estimate the most, and probably does not care about the fact that it is updated only after range data is added - it's an internal libcartographer detail. In that case I agree with both of your suggestions. 

On the other hand, if we do want to hint that this is closely related to processing range data, I suggest we go something with "range data" in the name.

Reply:
Let's call it `LocalSlamResultCallback` then.